pkg/storage/watch_cache.go

```diff
@@ -266,7 +266,7 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]wa
 		return result, nil
 	}
 	if resourceVersion < oldest {
-		return nil, errors.NewInternalError(fmt.Errorf("too old resource version: %d (%d)", resourceVersion, oldest))
+		return nil, errors.NewBadRequest(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest))
```

Comment:
Yeah I don't think it's a bad request. Also I *think* InternalError will give you a stack trace in apiserver logs, I would like to avoid this if it's the case. I suggest 410 if it avoids the stack trace.